graylog2-web-interface/src/components/common/URLWhiteListInput.tsx

```diff
@@ -31,12 +31,13 @@ type Props = {
   validationMessage: string,
   validationState: string,
   url: string,
+  setIsSubmitEnabled?: (enabled: boolean) => void,
   labelClassName: string,
   wrapperClassName: string,
   urlType: React.ComponentProps<typeof URLWhiteListFormModal>['urlType'],
 };
 
-const URLWhiteListInput = ({ label, onChange, validationMessage, validationState, url, labelClassName, wrapperClassName, urlType }: Props) => {
+const URLWhiteListInput = ({ label, onChange, validationMessage, validationState, url, setIsSubmitEnabled, labelClassName, wrapperClassName, urlType }: Props) => {
```

Comment:
While this approach works, I think it is breaking some of the semantics of the form: A form input should not decide if the submit button is enabled or disabled. This is also problematic when we expose this as our API, which is in the end what we do when we change the interface of pluggable entities.

I think the input could call a function once the validation state changes (maybe adding an `onValidationChange` prop that consumers can provide, or adding the validation state to the `onChange` callback). From my perspective, that would make it easier to understand, since it's a similar API one uses with another form, and decoupled from a particular implementation. What do you think?

Reply:
Hi, thank you for your review. I agree about the semantic issue. I will change that to use `onValidationChange.`